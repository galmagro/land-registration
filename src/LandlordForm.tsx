import React from 'react';
import {Breadcrumbs, Grid, Paper, TextField, Typography} from "@material-ui/core";
import {KeyboardDatePicker,} from '@material-ui/pickers';
import {Link} from "react-router-dom";
import { useTranslation } from 'react-i18next';

export default function LandlordForm() {

    const { t } = useTranslation();

    const onDateChange = (date: Date | null) => {
        setSelectedDate(date);
    }

    const [selectedDate, setSelectedDate] = React.useState<Date | null>();

    return (

            <Grid container spacing={2} style={{padding: 16}}>
                <Grid item xs={12}>
                    <Breadcrumbs>
                        <Link color="inherit" to="/landlords">{t("landlords")}</Link>
                        <Typography color="textPrimary">{t("new_landlord")}</Typography>
                    </Breadcrumbs>
                </Grid>
                <Grid item xs={12}>
                    <Typography variant="h3" style={{marginBottom: 20}}>
                        {t("new_landlord")}
                    </Typography>
                </Grid>
                <Grid item xs={6}>
                    <Grid container direction="column">
                        <Grid item xs={12}>
                            <form>
                                <Paper style={{padding: 16}} elevation={3}>

                                    <Typography variant="h6">{t("landlord_id_title")}</Typography>

                                    <TextField margin="normal" fullWidth required id="first_name"
                                               label={t("landlord_first_name")}/>

                                    <TextField margin="normal" fullWidth required id="last_name"
                                               label={t("landlord_last_name")}/>

                                    <KeyboardDatePicker variant="inline" onChange={onDateChange}
                                                        label={t("landlord_dob")}
                                                        value={selectedDate}
                                                        format="dd/MM/yyyy"
                                                        fullWidth
                                                        id="dob"
                                                        margin="normal"/>

                                    <TextField margin="normal" fullWidth required id="landlord_id"
                                                label={t("landlord_id")} placeholder={t("landlord_id_placeholder")}/>

                                    <TextField margin="normal" fullWidth required id="reference"
                                               label={t("landlord_ref")} placeholder={t("landlord_ref_placeholder")}/>
                                </Paper>
                            </form>
                        </Grid>
                    </Grid>
                </Grid>

                <Grid item xs={6}>

                </Grid>
            </Grid>
    );
}