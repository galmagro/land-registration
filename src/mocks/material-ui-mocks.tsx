import React from "react";

export const MockBreadcrumbs = (props: any) => <div className="Breadcrumbs">{props.children}</div>;
export const MockGrid = (props: any) => <div className="Grid">{props.children}</div>;
export const MockTextField = (props:any) => <input className="TextField" type="text" id={props.id}/>;
export const MockTypograghy = (props: any) => <span className="Typography">{props.children}</span>;
export const MockPaper = (props: any) => <div className="Paper">{props.children}</div>;

export const MockKeyboardDatePicker = (props: any) => <input type="date" id={props.id} className="KeyboardDatePicker"/>;
