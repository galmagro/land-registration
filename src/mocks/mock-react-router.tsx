import React from "react";

export const MockLink = (props: any) => <a href="#" className="Link">{props.children}</a>;