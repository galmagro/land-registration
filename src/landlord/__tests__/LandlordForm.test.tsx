import React from 'react';
import LandlordForm from "../LandlordForm";
import renderer from 'react-test-renderer';
import {
    MockBreadcrumbs,
    MockGrid, MockKeyboardDatePicker,
    MockPaper,
    MockTextField,
    MockTypograghy
} from "../../mocks/material-ui-mocks";
import {MockLink} from "../../mocks/mock-react-router";

jest.mock("@material-ui/core", () => {
    return {
        Breadcrumbs: (props: any) => <MockBreadcrumbs>{props.children}</MockBreadcrumbs>,
        Grid: (props: any) => <MockGrid>{props.children}</MockGrid>,
        Paper: (props: any) => <MockPaper>{props.children}</MockPaper>,
        TextField: (props: any) => <MockTextField id={props.id}/>,
        Typography: (props: any) => <MockTypograghy>{props.children}</MockTypograghy>
    }
});

jest.mock("@material-ui/pickers", () => {
    return {
        KeyboardDatePicker: (props: any) => <MockKeyboardDatePicker id={props.id}/>
    }
});

jest.mock("react-router-dom", () => {
    return {
        Link: (props: any) => <MockLink>{props.children}</MockLink>
    }
});

it('should compile correctly', () => {
    const component = renderer.create(<LandlordForm/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should have identification fields', () => {
    const component = renderer.create(<LandlordForm/>);
    const componentInstance = component.getInstance();
    expect(componentInstance?.findByProps({id: "first_name"})).not.toBeNull();
    expect(componentInstance?.findByProps({id: "last_name"})).not.toBeNull();
    expect(componentInstance?.findByProps({id: "landlord_dob"})).not.toBeNull();
    expect(componentInstance?.findByProps({id: "landlord_id"})).not.toBeNull();
    expect(componentInstance?.findByProps({id: "landlord_ref"})).not.toBeNull();
});
