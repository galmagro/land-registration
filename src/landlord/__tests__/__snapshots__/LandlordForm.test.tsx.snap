// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compile correctly 1`] = `
<div
  className="Grid"
>
  <div
    className="Grid"
  >
    <div
      className="Breadcrumbs"
    >
      <a
        className="Link"
        href="#"
      >
        landlords
      </a>
      <span
        className="Typography"
      >
        new_landlord
      </span>
    </div>
  </div>
  <div
    className="Grid"
  >
    <span
      className="Typography"
    >
      new_landlord
    </span>
  </div>
  <div
    className="Grid"
  >
    <div
      className="Grid"
    >
      <div
        className="Grid"
      >
        <form>
          <div
            className="Paper"
          >
            <span
              className="Typography"
            >
              landlord_id_title
            </span>
            <input
              className="TextField"
              id="first_name"
              type="text"
            />
            <input
              className="TextField"
              id="last_name"
              type="text"
            />
            <input
              className="KeyboardDatePicker"
              id="dob"
              type="date"
            />
            <input
              className="TextField"
              id="landlord_id"
              type="text"
            />
            <input
              className="TextField"
              id="reference"
              type="text"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="Grid"
  />
</div>
`;
